kolibri/content/views.py

```diff
@@ -40,8 +46,17 @@ def get(self, request, zipped_filename, embedded_filepath):
             # try to guess the MIME type of the embedded file being referenced
             content_type = mimetypes.guess_type(embedded_filepath)[0] or 'application/octet-stream'
 
-            # generate a streaming response object, pulling data from within the zip  file
-            response = FileResponse(zf.open(info), content_type=content_type)
+            if not os.path.splitext(embedded_filepath)[1] == '.json':
+                # generate a streaming response object, pulling data from within the zip  file
+                response = FileResponse(zf.open(info), content_type=content_type)
+                file_size = info.file_size
+            else:
+                # generate a streaming response object, pulling data from within the zip  file
+                content = zf.open(info).read()
+                str_to_be_replaced = str("\\n\\n![](/" + request.resolver_match.url_name + "/" + zipped_filename)
```

Comment:
Yeah, it involves changes on both sides. This will be necessary even for current Perseus stuff, since not all references files are images (there's graphie stuff, which includes both JSON and SVG files -- I can show you examples)